### PR TITLE
Standardize the logs in the eventing controller

### PR DESCRIPTION
**Pull request by VladislavPaskar**
**Description**

Changes proposed in this pull request:

- standardize the logs in the controller component
- fix the output in the cmd/main.go
- enhance some error messages
- remove redundant log messages

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/11681

### CODE REVIEW

**Comment by VladislavPaskar** [comment]
Considering the [comment ](https://github.com/kyma-project/kyma/issues/11681#issuecomment-1119484709), here is the difference now in the BEB reconciler:

old logs:
```shell
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"event-type-cleaner","caller":"eventtype/clean.go:56","message":"parse event-type failed","context":{"prefix":"sap.kyma.custom","type":"tunas.kyma.custom.noapp.order.created.v1","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-handler","caller":"handlers/beb.go:197","message":"clean application name in the subscription event-types failed","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","bebSubscriptionName":"test-noapp67a04fc2cefbff2c7387202d4cba82c42bd1e4bb","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-subscription-reconciler","caller":"beb/reconciler.go:275","message":"update BEB subscription failed","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-subscription-reconciler","caller":"beb/reconciler.go:148","message":"sync BEB subscription failed","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","caller":"controller/controller.go:326","message":"Reconciler error","context":{"controller":"beb-subscription-reconciler","object":{"name":"test-noapp","namespace":"tunas-testing"},"namespace":"tunas-testing","name":"test-noapp","reconcileID":"9994dd3e-0104-4170-82aa-79df9ec41af1","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"event-type-cleaner","caller":"eventtype/clean.go:56","message":"parse event-type failed","context":{"prefix":"sap.kyma.custom","type":"tunas.kyma.custom.noapp.order.created.v1","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-handler","caller":"handlers/beb.go:197","message":"clean application name in the subscription event-types failed","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","bebSubscriptionName":"test-noapp67a04fc2cefbff2c7387202d4cba82c42bd1e4bb","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-subscription-reconciler","caller":"beb/reconciler.go:275","message":"update BEB subscription failed","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","logger":"beb-subscription-reconciler","caller":"beb/reconciler.go:148","message":"sync BEB subscription failed","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:20:26Z","caller":"controller/controller.go:326","message":"Reconciler error","context":{"controller":"beb-subscription-reconciler","object":{"name":"test-noapp","namespace":"tunas-testing"},"namespace":"tunas-testing","name":"test-noapp","reconcileID":"c3ec9fe8-7614-4774-808e-7b0e10e930f8","error":"prefix not found"}}
```

new logs:
```shell
{"level":"ERROR","timestamp":"2022-07-01T08:15:39Z","logger":"beb-handler","caller":"handlers/beb.go:197","message":"Failed to clean application name in the subscription event-types","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","bebSubscriptionName":"test-noapp67a04fc2cefbff2c7387202d4cba82c42bd1e4bb","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:15:39Z","caller":"controller/controller.go:326","message":"Reconciler error","context":{"controller":"beb-subscription-reconciler","object":{"name":"test-noapp","namespace":"tunas-testing"},"namespace":"tunas-testing","name":"test-noapp","reconcileID":"bae00cf5-51f0-4622-9f8e-dd0aba56d845","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:16:00Z","logger":"beb-handler","caller":"handlers/beb.go:197","message":"Failed to clean application name in the subscription event-types","context"{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","bebSubscriptionName":"test-noapp67a04fc2cefbff2c7387202d4cba82c42bd1e4bb","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:16:00Z","caller":"controller/controller.go:326","message":"Reconciler error","context":{"controller":"beb-subscription-reconciler","object":{"name":"test-noapp","namespace":"tunas-testing"},"namespace":"tunas-testing","name":"test-noapp","reconcileID":"64a511e7-68ec-4090-a1d4-c195caff68ac","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:16:41Z","logger":"beb-handler","caller":"handlers/beb.go:197","message":"Failed to clean application name in the subscription event-types","context":{"kind":"Subscription","version":2,"namespace":"tunas-testing","name":"test-noapp","bebSubscriptionName":"test-noapp67a04fc2cefbff2c7387202d4cba82c42bd1e4bb","error":"prefix not found"}}
{"level":"ERROR","timestamp":"2022-07-01T08:16:41Z","caller":"controller/controller.go:326","message":"Reconciler error","context":{"controller":"beb-subscription-reconciler","object":{"name":"test-noapp","namespace":"tunas-testing"},"namespace":"tunas-testing","name":"test-noapp","reconcileID":"6c2b3c68-ef07-4b4e-be8c-28bdff7a2abc","error":"prefix not found"}}
```

File: components/event-publisher-proxy/cmd/event-publisher-proxy/main.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func main() {
 	}
 	defer func() {
 		if err := logger.WithContext().Sync(); err != nil {
-			golog.Fatalf("Failed to flush logger, error: %v", err)
+			golog.Printf("Failed to flush logger, error: %v", err)
```
Comment (VladislavPaskar) [inline components/event-publisher-proxy/cmd/event-publisher-proxy/main.go:62]: Reverting change done here https://github.com/kyma-project/kyma/pull/14712/files, we don't want the EPP to stop if the logger wasn't able to be flushed.

Comment (VladislavPaskar) [inline components/event-publisher-proxy/cmd/event-publisher-proxy/main.go:62]: Reverting the change done [here](https://github.com/kyma-project/kyma/pull/14712/files), b/c we don't want to stop the EPP if the logger failed to be flushed

File: components/eventing-controller/cmd/eventing-controller/main.go [hunk 2]
```diff
@@ -24,28 +25,27 @@ import (
...
 	ctrLogger, err := logger.New(opts.LogFormat, opts.LogLevel)
 	if err != nil {
-		setupLogger.Error(err, "initialize logger failed")
-		os.Exit(1)
+		log.Fatalf("Failed to initialize logger, error: %v", err)
 	}
 	defer func() {
 		if err := ctrLogger.WithContext().Sync(); err != nil {
-			setupLogger.Error(err, "flush logger failed")
+			log.Printf("Failed to flush logger, error: %v", err)
```
Comment (mfaizanse) [inline components/eventing-controller/cmd/eventing-controller/main.go:39]: Should it be `log.Errorf` ?

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 3]
```diff
@@ -168,19 +168,18 @@ func (r *Reconciler) reconcileNATSBackend(ctx context.Context, backendStatus *ev
 	if err != nil {
 		backendStatus.SetPublisherReadyCondition(false, eventingv1alpha1.ConditionReasonPublisherProxySecretError, err.Error())
 		if updateErr := r.syncBackendStatus(ctx, backendStatus, nil); updateErr != nil {
-			return ctrl.Result{}, errors.Wrapf(err, "update status when delete publisher proxy secret failed")
+			return ctrl.Result{}, errors.Wrapf(err, "failed to update status while deleting Event Publisher secret")
 		}
-		return ctrl.Result{}, errors.Wrapf(err, "delete eventing publisher proxy secret failed")
+		return ctrl.Result{}, errors.Wrapf(err, "delete eventing Event Publisher secret failed")
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/backend/reconciler.go:173]:
```
failed to delete....
``` 
??

File: components/eventing-controller/controllers/backend/reconciler.go [hunk 4]
```diff
@@ -620,7 +617,7 @@ func (r *Reconciler) CreateOrUpdatePublisherProxy(ctx context.Context, backend e
 
 	// Update publisher proxy deployment
 	if err := r.Update(ctx, desiredPublisher); err != nil {
-		return nil, errors.Wrapf(err, "update publisher proxy deployment failed")
+		return nil, errors.Wrapf(err, "update Event Publisher deployment failed")
```
Comment (mfaizanse) [inline components/eventing-controller/controllers/backend/reconciler.go:620]: `failed to...` ???

File: components/eventing-controller/pkg/handlers/jetstream.go [hunk 5]
```diff
@@ -533,7 +538,8 @@ func (js *JetStream) deleteConsumerFromJetStream(name string, log *zap.SugaredLo
 func (js *JetStream) checkJetStreamConnection(log *zap.SugaredLogger) error {
 	if js.conn.Status() != nats.CONNECTED {
 		if err := js.Initialize(js.connClosedHandler); err != nil {
-			log.Errorw("connect to JetStream failed", "status", js.conn.Status(), "error", err)
+			//todo check why double logging ?
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/handlers/jetstream.go:541]: Is this todo for you? or for follow-up for later?

Comment (VladislavPaskar) [reply components/eventing-controller/pkg/handlers/jetstream.go:541]: It was for me, removed it

File: components/eventing-controller/cmd/eventing-controller/main.go (hunk 2 above)
Comment (VladislavPaskar) [reply components/eventing-controller/cmd/eventing-controller/main.go:39]: Ideally yes, but the go-log doesn't have an equivalent to error, it has fatal which will also exit the program, the panic does the same. So the nearest one which will log the problem and continue the execution would be `Print`

File: components/eventing-controller/controllers/backend/reconciler.go (hunk 3 above)
Comment (VladislavPaskar) [reply components/eventing-controller/controllers/backend/reconciler.go:173]: I thought about it, but the thing is: the `errors.Wrapf` is essentially the error message, so according to the golang standards we shouldn't capitalise the error output. And the ticket was also only about the logging. I only touched the errors for consistency with the logs (like components naming) and for better sentence flow. But in general the error messages were not part of that PR.

File: components/eventing-controller/controllers/backend/reconciler.go (hunk 4 above)
Comment (VladislavPaskar) [reply components/eventing-controller/controllers/backend/reconciler.go:620]: https://github.com/kyma-project/kyma/pull/14725#discussion_r912035043

File: components/eventing-controller/cmd/eventing-controller/main.go (hunk 2 above)
Comment (mfaizanse) [reply components/eventing-controller/cmd/eventing-controller/main.go:39]: Can we then move this defer func definition after line 47, when we have an instance for `setupLogger`?

**Review by mfaizanse — APPROVED** [review]

**Pull request merged**